docs/buildrun.md

```diff
@@ -120,6 +120,10 @@ NAME                          SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
 buildpack-nodejs-buildrun     True        Succeeded   2m10s       74s
 ```
 
+### Build Snapshot
+
+For every BuildRun controller reconciliation, existing `BuildRun` resources(_Status_ field) are updated if an existing owned `TaskRun` is present. During this update, a `Build` resource snapshot is generated and embedded into the `status.buildSpec` path of the `BuildRun`. A `buildSpec` is just a copy of the original `Build` spec, from where the `BuildRun` executed a particular image build. The snapshot approach allows developers to see the original `Build` configuration.
```

Comment:
Thanks for the doc improvement.

few comments:

> For every BuildRun controller reconciliation, existing BuildRun resources(Status field) are updated if an existing owned TaskRun is present.

I think we can refine it more from end-user perspective to avoid the confusion:
- For `Build Snapshot`, the status is just updated when `TaskRun` is present, not for every BuildRun controller reconciliation
- For `All statuses` include the Build Snapshot status, the status will be updated differently, not just when `TaskRun` is present. 
- I think we can just say the Status of the existing `BuildRun` instead of `existing BuildRun resources(_Status_ field)`

` A buildSpec is just a copy of the original Build spec`:
We may also change it not just the copy, we may also fetch the real commit id and save in the buildSpec status: https://github.com/redhat-developer/build/issues/205

So I think we can keep the statement more sample first.

Reply:
> For Build Snapshot, the status is just updated when TaskRun is present, not for every BuildRun controller reconciliation

yes, I think this is why I wrote `if an existing owned TaskRun is present.` . So if you reconcile 10 times, and the TaskRun is there, it gets updated.

> For All statuses include the Build Snapshot status, the status will be updated differently, not just when TaskRun is present.

I think the sentence just speak about when you reconcile, and a taskrun is there, basically https://github.com/redhat-developer/build/blob/master/pkg/controller/buildrun/buildrun_controller.go#L126-L148

> I think we can just say the Status of the existing BuildRun instead of existing BuildRun resources(_Status_ field)

Yes, changing.

> A buildSpec is just a copy of the original Build spec:
is not the case now, so not need to write something is not happening.

Reply:
Discussed with Enrique and refine the words.

Thanks Enrique!